### ISSUE TITLE
R.player.playingSource() should receive playlist changes

### ISSUE THREAD

**Issue by matelich** [rdio/api#122]
When playing a playlist, changes to the playlist do not get loaded in any reasonable amount of time (many minutes).  removeFromPlaylist, addToPlaylist, setPlaylistOrder - none of them trigger the cached tracks to update.

The only way I know of to get an update is to delete a track less than the current index _and_ that only works if I am not the master.  This is not 100% reliable, but pretty reliable.

**Comment by iangilman** [comment]
Yes, this is an issue on rdio.com as well... once you start playing a playlist, the player is not guaranteed to get the changes you make to it. I've filed a bug for it and will update here when it gets fixed. 

Out of curiosity, what are you trying to accomplish? Did you just happen to be playing a playlist while editing it and notice that your changes weren't being updated in playingSource() or does your app revolve around this scenario?

**Comment by matelich** [comment]
Yes, my app revolves around this.  Its an album radio site.  Deletes played songs from the front of a playlist, adds albums at the end, supports delete and reorder of the albums.

http://matelich.github.io/album-radio/

The rdio site is much more responsive to changes.  The playlist page must not be using the tracks from playingSource, but access the playlist directly.

**Comment by iangilman** [comment]
Discussed this some more with @matelich on IRC. For the record here, other options include not using shared_playstate and using R.player.queue. If you need shared_playstate, you can use:

https://github.com/rdio/jsapi-examples/blob/master/utils/README.md#rdioutilslocalqueue-config-

Of course we'll also update here when the bug gets fixed or there's another alternate way to accomplish this.
